polus-training-splinedist-plugin/src/utils.py

```diff
@@ -0,0 +1,520 @@
+
+        # The corresponding label for the image
+        br_label = BioReader(label)
+        lab_array = br_label[:,:,0:1,0:1,0:1]
+        lab_array = lab_array.reshape(br_label.shape[:2])
+        array_labels_tested.append(fill_label_holes(lab_array))
+        
+        assert im_array.shape == lab_array.shape, "{} and {} do not have matching shapes".format(base_image, base_label)
+
+        if (i%tenpercent_tested == 0) and (i!=0):
+            logger.info("Loaded ~{}% of Test Data".format(np.ceil((i/num_tested)*100), num_tested))
+    logger.info("Done Loading Testing Data")
+
+    # Read the input images used for training
+    logger.info("\n Starting to Load Train Data ...")
+    tenpercent_trained = np.floor(num_trained/10)
+    contoursize_max = 0 # gets updated if models have not been created for config file
+    for i in range(num_trained):
+
+        br_image = BioReader(X_trn[i])
+        br_image_shape = br_image.shape[:2]
+        im_array = br_image[:,:,0:1,0:1,0:1]
+        im_array = im_array.reshape(br_image_shape)
+        array_images_trained.append(normalize(im_array,pmin=1,pmax=99.8,axis=axis_norm))
+
+        if i == 0:
+            n_channel = br_image.shape[2]
+
+        br_label = BioReader(Y_trn[i])
+        lab_array = br_label[:,:,0:1,0:1,0:1]
+        lab_array = lab_array.reshape(br_label.shape[:2])
+        array_labels_trained.append(fill_label_holes(lab_array))
+        
+        assert im_array.shape == lab_array.shape, "{} and {} do not have matching shapes".format(base_image, base_label)
+        contoursize_max = update_countoursize_max(contoursize_max, lab_array)
+        
+        if (i%tenpercent_trained == 0) and (i!=0):
+            logger.info("Loaded ~{}% of Train Data -- contoursize_max: {}".format(np.ceil((i/num_trained)*100), contoursize_max))
+    logger.info("Done Loading Training Data")
+        
+
+    # Build the model and generate other necessary files to train the data.
+    logger.info("Max Contoursize: {}".format(contoursize_max))
+
+    n_params = M*2
+    grid = (2,2)
+    conf = Config2D (
+    n_params            = n_params,
+    grid                = grid,
+    n_channel_in        = n_channel,
+    contoursize_max     = contoursize_max,
+    train_epochs        = epochs,
+    use_gpu             = gpu
+    )
+    
+    del X_val
+    del Y_val
+    del X_trn
+    del Y_trn
+
+    logger.info("\n Generating phi and grids ... ")
+    if not os.path.exists("./phi_{}.npy".format(M)):
+        phi_generator(M, conf.contoursize_max, '.')
+    logger.info("Generated phi")
+    if not os.path.exists("./grid_{}.npy".format(M)):
+        grid_generator(M, conf.train_patch_size, conf.grid, '.')
+    logger.info("Generated grid")
+
+    model = SplineDist2D(conf, name=model_dir_name, basedir=output_directory)
+    del conf
+
+    # After creating or loading model, train it
+    # model.config.train_tensorboard = False
+    # model.config.use_gpu = gpu
```

Comment:
Is there a reason to keep this commented code?

Reply:
No, I forgot to delete it.  The gpu was using too much memory when I was trying to build my models